### PR TITLE
[CPU]Interpolate extend to rank 3 input support

### PR DESCRIPTION
**Pull request by chenhu-wang**
### Details:
 - *extend to support rank 3 input*

### Tickets:
 - *CVS-123800*

### CODE REVIEW

**Comment by dmitry-gorokhov** [comment]
@xuchen-intel please review the PR

**Review by xuchen-intel — APPROVED** [review]
LGTM. Just one minor comment, can we also remove the local variable dataRank in Interpolate::isSupportedOperation, as is done in Interpolate::Interpolate?

**Comment by chenhu-wang** [comment]
Regarding: Just one minor comment, can we also remove the local variable dataRank in Interpolate::isSupportedOperation, as is done in Interpolate::Interpolate?

Thanks for your suggestion. However, Interpolate::isSupportedOperation is a static function. 1. it's called before Interpolate::Interpolate. 2. static function can not access non-static member. 
What's more, isSupportedOperation should be separate from cpu node implementation. There are possibility that some transformation happens between them, and isSupportedOperation maybe not used at all for some nodes. So I prefer to leave it.

**Review by nikita-savelyevv — APPROVED** [review]

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp [hunk 1]
```diff
@@ -349,6 +329,244 @@ std::vector<std::map<std::string, std::string>> filterAdditionalConfig() {
...
+    } else {
+        resCPUParams.push_back(CPUSpecificParams{{ncw, x, x, x}, {ncw}, {"ref"}, "ref"});
+    }
+    return resCPUParams;
+}
+
+std::vector<std::map<std::string, std::string>> filterAdditionalConfig3D() {
+    return {
+        // default config as an stub
+        {{InferenceEngine::PluginConfigParams::KEY_PERF_COUNT, InferenceEngine::PluginConfigParams::NO}}
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp:346]: Why exactly this option? Just leave empty config.

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp [hunk 2]
```diff
@@ -349,6 +329,244 @@ std::vector<std::map<std::string, std::string>> filterAdditionalConfig() {
     }
 }
 
+// 3D
```
Comment (dmitry-gorokhov) [inline src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp:332]: How many new test-cases we add here? I'd like to minimize impact on functional test pass time

**Comment by dmitry-gorokhov** [comment]
@chenhu-wang please rebase the PR on latest master to get fresh CI

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp (hunk 1 above)
Comment (chenhu-wang) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp]: I thought empty vector lead to zero test cases(0*x*y..) previously. I have tested that empty just not set config, it's still a stub.  Have updated, thanks Dmitry.

File: src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp (hunk 2 above)
Comment (chenhu-wang) [reply src/plugins/intel_cpu/tests/functional/single_layer_tests/interpolate.cpp:332]: 624 cases are added for 3D smoke test, which toke 8.5s. I have mitigated the number by 1) move non-zero pad to post-commit, which is not common. 2) reduce input shapes enumeration. Now there are **96 cases, taking 1.3s** for smoke case. 502 cases for post-commit, take 6.8s. @Dmitry, Do you think it is acceptable? thanks!

**Comment by chenhu-wang** [comment]
> @chenhu-wang please rebase the PR on latest master to get fresh CI

rebased.

**Review by dmitry-gorokhov — APPROVED** [review]

**Pull request merged**